### PR TITLE
Use types from common.h

### PR DESCRIPTION
**Pull request by Broxzier**
Because of a comment on a diff in https://github.com/OpenRCT2/OpenRCT2/pull/4878 asking for clearer and more consistent type usage. What this PR changes:

No `unsigned long long` found
`(signed) long long` -> `sint64`
`unsigned int` -> `uint32`
`(signed) int` -> `sint32`
`(signed) long` -> `sint32`
`unsigned short` -> `uint16`
`(signed) short` -> `sint16`
`unsigned char` -> `uint8`
No `signed char`s found
plain `char` is implementation-defined, so I left those in.

I saw some `unsigned long`s being used for things like buffersizes, lengths, offsets, and array access, shall I replace them with `size_t`?

### CODE REVIEW

**Comment by IntelOrca** [comment]
Use our own defined types: `uint32` instead of `uint32_t`

**Comment by IntelOrca** [comment]
yes

File: src/cmdline/CommandLine.cpp [hunk 1]
```diff
@@ -280,7 +280,7 @@ namespace CommandLine
 
         switch (option->Type) {
         case CMDLINE_TYPE_INTEGER:
-            String::Append(buffer, bufferSize, "=<int>");
+            String::Append(buffer, bufferSize, "=<sint32>");
```
Comment (marijnvdwerf) [inline src/cmdline/CommandLine.cpp]: Is this string supposed to get changed?

File: src/core/MemoryStream.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ class MemoryStream final : public IStream
     uint64  GetLength()                               const override;
     uint64  GetPosition()                             const override;
     void    SetPosition(uint64 position)                    override;
-    void    Seek(sint64 offset, int origin)                 override;
+    void    Seek(sint64 offset, sint32 origin)                 override;
```
Comment (marijnvdwerf) [inline src/core/MemoryStream.h]: alignment

File: src/drawing/sprite.c [hunk 3]
```diff
@@ -360,10 +360,10 @@ void FASTCALL gfx_draw_sprite_software(rct_drawpixelinfo *dpi, int image_id, int
 * x (cx)
 * y (dx)
 */
-void FASTCALL gfx_draw_sprite_palette_set_software(rct_drawpixelinfo *dpi, int image_id, int x, int y, uint8* palette_pointer, uint8* unknown_pointer)
+void FASTCALL gfx_draw_sprite_palette_set_software(rct_drawpixelinfo *dpi, sint32 image_id, sint32 x, sint32 y, uint8* palette_pointer, uint8* unknown_pointer)
```
Comment (marijnvdwerf) [inline src/drawing/sprite.c]: Why are image id's signed?

File: src/cmdline/CommandLine.cpp (hunk 1 above)
Comment (Broxzier) [inline src/cmdline/CommandLine.cpp]: Well spotted.

File: src/core/MemoryStream.h (hunk 2 above)
Comment (Broxzier) [inline src/core/MemoryStream.h]: Doesn't the clang-format tool fix this automatically? There are many more of those cases.

File: src/drawing/sprite.c (hunk 3 above)
Comment (Broxzier) [inline src/drawing/sprite.c]: No idea, it's been that way since https://github.com/OpenRCT2/OpenRCT2/commit/b828f678 apparently.

**Comment by marijnvdwerf** [comment]
I still find it pretty difficult to get through this, and Chrome is struggling to show the diff page. I'm not sure about the ride paint files, but I think vanilla treated those as `uint16`. What about moving those ride files to a separate PR, so things can be merged in more bite-sized parts?

**Comment by janisozaur** [comment]
@Broxzier sorry, I had to cancel appveyor on this PR. The first job came green, which should indicate remaining 3 also pass, but travis was red, so I expect more commits here in the future anyway.

We clogged the CIs today and I needed to do something about it.

**Comment by IntelOrca** [comment]
@Broxzier its better waiting for #4932 to be merged, otherwise you will end up with merge hell.

**Comment by Broxzier** [comment]
@janisozaur: It's not a problem.
@IntelOrca: Sure thing, I wasn't aware of those changes when I started, but it's easy to redo.

File: src/openrct2/diagnostic.h [hunk 4]
```diff
@@ -73,14 +75,14 @@ enum {
...
-extern int _log_levels[DIAGNOSTIC_LEVEL_COUNT];
+extern sint32 _log_levels[DIAGNOSTIC_LEVEL_COUNT];
 
 #ifdef __cplusplus
 extern "C" {
 #endif // __cplusplus
 
-void diagnostic_log(int diagnosticLevel, const char *format, ...);
-void diagnostic_log_with_location(int diagnosticLevel, const char *file, const char *function, int line, const char *format, ...);
+void diagnostic_log(sint32 diagnosticLevel, const char *format, ...);
```
Comment (marijnvdwerf) [inline src/openrct2/diagnostic.h]: `typedef` the `DIAGNOSTIC_LEVEL_...` enum instead?

File: src/openrct2/diagnostic.h [hunk 5]
```diff
@@ -17,6 +17,8 @@
 #ifndef _DIAGNOSTIC_H_
 #define _DIAGNOSTIC_H_
 
+#include "common.h"
+
 enum {
```
Comment (marijnvdwerf) [inline src/openrct2/diagnostic.h]: Why not `typdef enum {} DiagnosticLevel`?

Comment (Broxzier) [inline src/openrct2/diagnostic.h]: I wasn't sure that's valid in both C and C++.

**Comment by janisozaur** [comment]
@Broxzier You shouldn't fox a Linux.

The CIs won't pick an unmergable change, so you need to fix that first.

**Comment by Broxzier** [comment]
@Gymnasiast Rebased now, and changed a few instances from sint32/uint32 to size_t where appropriate.
@janisozaur You just have to believe!

**Comment by IntelOrca** [comment]
https://ci.appveyor.com/project/IntelOrca/openrct2-ject9/build/0.0.6.6763/job/5pk4ncnbwnc1rg1v/messages

**Comment by IntelOrca** [comment]
One problem with this PR is that I presume you have done a find and replace? so it has broken a lot of code code alignment.

**Comment by janisozaur** [comment]
@IntelOrca that's why people use `clang-format`, to not be bothered by such issues.

**Comment by Broxzier** [comment]
@IntelOrca: Yes, it's mostly find and replace. I'm not getting these warnings in VS though.

**Comment by IntelOrca** [comment]
@Broxzier they are warnings for 32-bit only

**Comment by Broxzier** [comment]
@IntelOrca I am building in 32-bit. But I just noticed some warnings have been disabled in the project settings.

**Comment by IntelOrca** [comment]
@janisozaur does clang-format align code vertically?

**Comment by janisozaur** [comment]
@IntelOrca https://github.com/OpenRCT2/OpenRCT2/pull/5054/files

**Comment by Broxzier** [comment]
@IntelOrca: AlignConsecutiveAssignments https://github.com/Microsoft/clang/commit/aea0264e0cb2c6c3fcb3f0f4c2f723f379940468

**Comment by janisozaur** [comment]
@Broxzier @IntelOrca https://github.com/OpenRCT2/OpenRCT2/blob/3d7cfa2/.clang-format#L4-L5

**Comment by Broxzier** [comment]
The warnings should be fixed now.

**Comment by IntelOrca** [comment]
Could you go through every `class` and `interface` and fix the indentation?

**Comment by Broxzier** [comment]
@IntelOrca https://github.com/OpenRCT2/OpenRCT2/pull/5071 is already taking care of that. If you merge this one first, then https://github.com/OpenRCT2/OpenRCT2/pull/5071 it should be alright.

**Comment by IntelOrca** [comment]
#5071 is not a good idea

**Pull request merged**